.github/workflows/macos_tests.yml

```diff
@@ -0,0 +1,82 @@
+name: MacOS tests
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+  release:
+  workflow_dispatch:
+
+jobs:
+  tests:
+    name: Test - ${{ matrix.os }} - Build type ${{ matrix.build_type }} - Build testing ${{ matrix.build_testing }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - macos-11
+          - macos-12
+          # TODO: can change to macos-latest (but it's currently aliased to macos-11)
+        build_type:
+          - Release
+        build_testing:
+          - ON
+          - OFF
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v3
+
+      - name: Setup cache
+        id: cache-test-data
+        uses: actions/cache@v3
+        with:
+          path: ${{ github.workspace }}/tdms/tests/system/data
+          key: test_data
+
+      - name: Set up MATLAB
+        uses: matlab-actions/setup-matlab@v1
+
+      - name: Install dependencies
+        run: |
+          brew install fftw llvm libomp python3
+
+      - name: Install Python dependencies
+        shell: bash
+        run: |
+          pip3 install -r ${GITHUB_WORKSPACE}/tdms/tests/requirements.txt
+
+      - name: Create Build Environment
+        run: |
+          cmake -E make_directory ${{runner.workspace}}/build
+
+      - name: Configure TDMS
+        shell: bash
+        working-directory: ${{runner.workspace}}/build
+        env:
+          CXXFLAGS: -Werror
+        run: |
+          cmake ${GITHUB_WORKSPACE}/tdms -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON\
+           -DCMAKE_BUILD_TYPE=${{ matrix.build_type }}\
+           -DBUILD_TESTING=${{ matrix.build_testing }}\
+           -DOMP_ROOT=/usr/local/opt/llvm/ -DCXX_ROOT=/usr/local/opt/llvm
+
+      - name: Build TDMS
+        working-directory: ${{runner.workspace}}/build
+        shell: bash
+        run: |
+          cmake --build . --config ${{ matrix.build_type }}
+
+      - name: Run TDMS unit tests
+        if: matrix.build_testing == 'ON'
+        working-directory: ${{runner.workspace}}/build
+        run: |
+          export OMP_NUM_THREADS=1
```

Comment:
Instead of setting `OMP_NUM_THREADS` manually in each step (and I'd use
```yaml
        env:
          OMP_NUM_THREADS: 1
```
instead), I think you can also set the variable globally for the entire job.  This has also the advantage of being shell-independent (not that on macOS specifically we'd use much different shells, but I'm thinking in the optic of unifying the build scripts).

Reply:
tangential to this point: it may be nice to have a test e.g. just `test_arc_01.py `that runs with 2 threads, to test the parallelisation hasn't been broken?

Reply:
Sounds good.  It could be done with an additional matrix entry, and then you can do
```yaml
env:
  OMP_NUM_THREADS: ${{ matrix.num_threads }}
```
We do it for example at https://github.com/Team-RADDISH/ParticleDA.jl/blob/410605a29b17d2ec2f3e86848e62a4ab6ff73102/.github/workflows/ci.yml#L12-L13

Reply:
> Instead of setting `OMP_NUM_THREADS` manually in each step (and I'd use
> 
> ```yaml
>         env:
>           OMP_NUM_THREADS: 1
> ```
> 
> instead), I think you can also set the variable globally for the entire job. This has also the advantage of being shell-independent (not that on macOS specifically we'd use much different shells, but I'm thinking in the optic of unifying the build scripts).

Ah. This was just brainlessly copied over from the Wndows one which was the starting point.

Reply:
> Sounds good. It could be done with an additional matrix entry, and then you can do
> 
> ```yaml
> env:
>   OMP_NUM_THREADS: ${{ matrix.num_threads }}
> ```
> 
> We do it for example at https://github.com/Team-RADDISH/ParticleDA.jl/blob/410605a29b17d2ec2f3e86848e62a4ab6ff73102/.github/workflows/ci.yml#L12-L13

I would do this in a separate issue/PR (to avoid snowballing). And let's do it to the linux tests (only)?